clientv3/integration/user_test.go

```diff
@@ -51,3 +52,51 @@ func TestUserError(t *testing.T) {
 		t.Fatalf("expected %v, got %v", rpctypes.ErrRoleNotFound, err)
 	}
 }
+
+func TestUserErrorAuth(t *testing.T) {
+	defer testutil.AfterTest(t)
+
+	clus := integration.NewClusterV3(t, &integration.ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+
+	authapi := clus.RandClient()
+	authSetupRoot(t, authapi.Auth)
+
+	// un-authenticated client
+	if _, err := authapi.UserAdd(context.TODO(), "foo", "bar"); err != rpctypes.ErrUserNotFound {
+		t.Fatalf("expected %v, got %v", rpctypes.ErrUserNotFound, err)
+	}
+
+	// wrong id or password
+	if _, err := integration.NewClientV3Auth(clus.Members[0], "foo", "bar"); err != rpctypes.ErrAuthFailed {
```

Comment:
can this not add any new functions to `integration`? it seems like if it's defining auth stuff the client should be created off a user-supplied config...

```
cli := clus.Client(0)
cfg := clientv3.Config{Endpoints: cli.Endpoints()}
cfg.Username, cfg.Password = "foo", "bar" 
if _, err := clientv3.New(cfg); err != rpctypes.ErrAuthFailed {
...
}
...
```